src/harness/testcases/WINNF_FT_S_GRA_testcase.py

```diff
@@ -276,6 +276,232 @@ def test_WINNF_FT_S_GRA_4(self):
+                                config['sasTestHarnessConfig']['serverCert'],
+                                config['sasTestHarnessConfig']['serverKey'],
+                                config['sasTestHarnessConfig']['caCert'])
+    sas_test_harness_server.writeFadRecords(sas_test_harness_dump_records)
+
+    # Start the SAS Test Harness server
+    sas_test_harness_server.start()
+
+    # Notify the SAS UUT about the SAS Test Harness
+    certificate_hash = getCertificateFingerprint(config['sasTestHarnessConfig']['serverCert'])
+    self._sas_admin.InjectPeerSas({'certificateHash': certificate_hash,
+                                  'url': sas_test_harness_server.getBaseUrl()})
+
+    # Step 2: Trigger CPAS in the SAS UUT and wait until complete.
+    self.TriggerDailyActivitiesImmediatelyAndWaitUntilComplete()
+
+    # Pre-load conditional registration data for C1 and C2 CBSDs.
+    if ('conditionalRegistrationData' in config) and (
+        config['conditionalRegistrationData']):
+      self._sas_admin.PreloadRegistrationData(
+          config['conditionalRegistrationData'])
+
+    # Step 3: Register CBSDs C1 and C2 with SAS UUT
+    # The assertRegistered function does the Inject FCC ID and user ID for the registration requests
+    cbsd_ids = self.assertRegistered([device_c1, device_c2])
+    grant_g1['cbsdId'] = cbsd_ids[0]
+    grant_g2['cbsdId'] = cbsd_ids[1]
+
+    # Step 4: Send a valid Grant Request for C1 and C2 to the SAS UUT.
+    request = {'grantRequest': [grant_g1, grant_g2]}
+    response = self._sas.Grant(request)['grantResponse']
+
+    # Check the length of request and response match.
+    self.assertEqual(len(request['grantRequest']), len(response))
+
+    # Check responseCode is 401 for CBSD C1.
+    self.assertEqual(response[0]['response']['responseCode'], 401)
+
+    # Check responseCode is 0 for CBSD C2.
+    self.assertEqual(response[1]['response']['responseCode'], 0)
+
+    # As Python garbage collector is not very consistent, directory is not getting deleted. 
+    # Hence, explicitly stopping SAS Test Hanress and cleaning up
+    sas_test_harness_server.shutdown()
+    del sas_test_harness_server
+
+  def generate_GRA_6_default_config(self, filename):
+    """Generates the WinnForum configuration for GRA_6"""
+    # Create the actual config for GRA_6
+
+    # Load device_a with registration in SAS Test Harness.
+    device_c1 = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_a.json')))
+
+    # Load grant request.
+    grant_g1 = json.load(
+        open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+
+    grant_g2 = json.load(
+        open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    grant_g2['operationParam']['operationFrequencyRange'][
+        'lowFrequency'] = 3645000000
+    grant_g2['operationParam']['operationFrequencyRange'][
+        'highFrequency'] = 3655000000
+
+    cbsd_record = [device_c1]
```

Comment:
Please use plural to indicate that these are lists.

Reply:
changed to plural for all applicable cases.